docs/usage/shoot-operations/shoot_updates.md

```diff
@@ -109,6 +140,127 @@ Changes to `kubeReserved` or `systemReserved` do not trigger a node roll if thei
+
+#### InPlace Update Triggers
+
+An in-place update of the shoot worker nodes is triggered for rolling update triggers listed under [Rolling Update Triggers](#rolling-update-triggers) except the following:
+* `.spec.provider.workers[].machine.image.name`
+* `.spec.provider.workers[].machine.type`
+* `.spec.provider.workers[].volume.type`
+* `.spec.provider.workers[].volume.size`
+* `.spec.provider.workers[].cri.name`
+* `.spec.systemComponents.nodeLocalDNS.enabled`
+
+#### Validations for In-Place update
+
+When a worker pool is undergoing an in-place update, applying subsequent updates to the same worker pool is restricted.
+If an in-place update fails and nodes are left in a problematic state, user intervention is required to manually fix the nodes. After resolving the issue, users can restart the update by adding the force update annotation `gardener.cloud/operation=force-in-place-update`.
+
+Without this annotation, subsequent updates to the same worker pool are denied. Refer to [Force-update a worker pool with InPlace update strategy](shoot_operations.md#force-update-a-worker-pool-with-inplace-update-strategy) for more details.
+
+> ⚠️ Changing the update strategy from `AutoRollingUpdate` to `AutoInPlaceUpdate` or `ManualInPlaceUpdate` (and vice versa) is not allowed. However, switching between `AutoInPlaceUpdate` and `ManualInPlaceUpdate` is permitted.
+
+#### AutoInPlaceUpdate
+
+In case of AutoInPlaceUpdate update strategy, the update process is fully orchestrated by Gardener and the machine-controller-manager. No user intervention is required.The update strategy should be set to `AutoInPlaceUpdate` in the worker spec.
+
+The `inPlaceUpdates.pendingWorkerUpdates.autoInPlaceUpdate` field in the Shoot status lists the names of worker pools that are pending updates with the `AutoInPlaceUpdate` strategy.
+
+```yaml
+spec:
+  provider:
+    type: <some-provider-name>
+    workers:
+      - name: cpu-worker
+        minimum: 5
+        maximum: 5
+        maxSurge: 0
+        maxUnavailable: 2
+        updateStrategy: AutoInPlaceUpdate
+        machine:
+          type: m5.large
+          image:
+            name: <some-image-name>
+            version: <some-image-version>
+          architecture: <some-cpu-architecture>
+        providerConfig: <some-machine-image-specific-configuration>
+```
+
+#### Customize Auto In-Place Update Behaviour of Shoot Worker Nodes
+
+The `.spec.provider.workers[]` list exposes two fields that you might configure based on your workload's needs: `maxSurge` and `maxUnavailable`.
+The same concepts [like in Kubernetes](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#rolling-update-deployment) apply.
+Additionally, you might customize how the machine-controller-manager is behaving. You can configure the following fields in `.spec.provider.worker[].machineControllerManager`:
```

Comment:
Refer here the list of other MCM configuration which is listed for AutoRollingUpdate.